### PR TITLE
fix temp file issue

### PR DESCRIPTION
**Pull request by cloudmagic80**
## Summary
feat: Ensure unique template file names and fix file path issues

- Updated `generate` method to construct `generated_form_path` and `stamped_template_path` using `Rails.root.join` for better path handling.
- Added `create_tempfile` method to create a temporary file for the template, ensuring unique file names by including `SecureRandom.hex`.
- Modified `create_tempfile` method to use `Tempfile` for creating temporary files and copying the template file to the temporary file.
- Ensured that the `template_form_path` is correctly constructed using `TEMPLATE_BASE` and `form_number`.
- Added error handling to raise an exception if the stamped template file does not exist after copying.

## Related issue(s)

https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/simple_forms_api/app/services/simple_forms_api/pdf_filler.rb

## Testing done

7959f1 and 1010d

## What areas of the site does it impact?
All ivc champva forms

### CODE REVIEW

**Review by michaelclement — APPROVED** [review]

File: modules/ivc_champva/app/services/ivc_champva/pdf_filler.rb
Comment (rmtolmach) [inline modules/ivc_champva/app/services/ivc_champva/pdf_filler.rb:1]: Can you add unit tests?

File: modules/ivc_champva/app/services/ivc_champva/pdf_filler.rb [hunk 1]
```diff
@@ -18,10 +19,12 @@ def initialize(form_number:, form:, name: nil)
     end
 
     def generate(current_loa = nil)
-      template_form_path = "#{TEMPLATE_BASE}/#{form_number}.pdf"
-      generated_form_path = "tmp/#{name}-tmp.pdf"
-      stamped_template_path = "tmp/#{name}-stamped.pdf"
-      FileUtils.copy(template_form_path, stamped_template_path)
+      generated_form_path = Rails.root.join("tmp/#{name}-#{SecureRandom.hex}-tmp.pdf").to_s
```
Comment (acrollet) [inline modules/ivc_champva/app/services/ivc_champva/pdf_filler.rb:22]: Is there any reason we couldn't do something like this here and keep things a little simpler? `Tempfile` already guarantees a unique filename (or exception)

`generated_form_path = Tempfile.new(name).path`

Comment (cloudmagic80) [reply modules/ivc_champva/app/services/ivc_champva/pdf_filler.rb:22]: Thanks for the suggestion! The reason I'm using Rails.root.join with a specific filename pattern is that we want to ensure the generated PDF path remains in the tmp folder of the project directory. Using Tempfile.new would place the file in the system's temp directory (e.g., /var/folders/...), which differs from our desired structure (tmp/...). This approach also makes it easier to locate and manage temporary files within the project scope.

File: modules/ivc_champva/app/services/ivc_champva/pdf_filler.rb
Comment (cloudmagic80) [reply modules/ivc_champva/app/services/ivc_champva/pdf_filler.rb:1]: Completed!

File: modules/ivc_champva/app/services/ivc_champva/pdf_filler.rb (hunk 1 above)
Comment (acrollet) [reply modules/ivc_champva/app/services/ivc_champva/pdf_filler.rb:22]: Tempfile also allows passing a directory, but I don't feel that strongly about this and don't want to hold the PR up. The only motivation is to have simpler code :-)

**Review by reiting — APPROVED** [review]

**Review by rmtolmach — APPROVED** [review]

**Pull request merged**